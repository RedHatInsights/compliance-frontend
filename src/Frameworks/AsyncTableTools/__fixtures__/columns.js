import { severityLevels } from './items';

export default [
  {
    title: 'ID',
    sortByProperty: 'id',
    exportKey: 'id',
    key: 'id',
    manageable: false,
  },
  {
    title: 'Name',
    key: 'id',
    sortByProperty: 'name',
    exportKey: 'name',
    renderFunc: (_a, _b, item) => item.name,
  },
  {
    title: 'Another Name column',
    key: 'another_name_column',
    sortByProperty: 'name',
    exportKey: 'name',
    renderExport: (name) => `${name} via export render`,
  },
  {
    title: 'Description',
    key: 'desc',
    sortByProperty: 'description',
    renderExport: (item) => `${item.name} description rendered for export`,
  },
  {
    title: 'Severity sorted by function',
    sortByFunction: (item) => item.name,
    exportKey: 'severity',
    renderFunc: (_a, _b, item) => item.severity,
  },
  {
    title: 'Severity by Array',
    sortByProperty: 'severity',
    sortByArray: severityLevels,
    exportKey: 'severity',
    renderFunc: (_a, _b, item) => item.severity,
  },
  {
    title: 'Download button',
    key: 'download',
    manageable: false,
    export: false,
    renderFunc: () => 'download link',
  },
];
