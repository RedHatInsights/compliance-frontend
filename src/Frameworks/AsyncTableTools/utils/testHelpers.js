import React from 'react';
import TableStateProvider from '../components/TableStateProvider';

export const createTestWrapper = (Wrapper, props) => {
  return function CreatedWrapper({ children }) { // eslint-disable-line
    return <Wrapper {...props}>{children}</Wrapper>;
  };
};

export const DEFAULT_RENDER_OPTIONS = {
  wrapper: createTestWrapper(TableStateProvider),
};
