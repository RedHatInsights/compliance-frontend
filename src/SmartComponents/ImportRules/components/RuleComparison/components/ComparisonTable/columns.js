import React from 'react';
import { nowrap } from '@patternfly/react-table';

import RuleCell from './components/RuleCell';
import ProfileVersionCell from './components/ProfileVersionCell';

const rule = {
  title: <strong>Rule</strong>,
  Component: RuleCell,
};

export const profileVersion = (version, targetVersion, additionalProps) => ({
  dataLabel: `RHEL ${version.os_major_version}.${version.os_minor_version} -  SSG version ${version.ssg_version}`,
  title: (
    <>
      <strong>
        RHEL {version.os_major_version}.{version.os_minor_version}
      </strong>
      <br />
      SSG version {version.ssg_version}
    </>
  ),
  Component: (props) => (
    <ProfileVersionCell
      {...{
        version,
        targetVersion,
        ...props,
        ...additionalProps,
      }}
    />
  ),
  transforms: [nowrap],
});

export default [rule];
