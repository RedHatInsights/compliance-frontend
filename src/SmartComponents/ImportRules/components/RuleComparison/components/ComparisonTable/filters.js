import { conditionalFilterType } from '@redhat-cloud-services/frontend-components/ConditionalFilter';

const rule = {
  type: conditionalFilterType.text,
  label: 'Rule',
  filterSerialiser: (_, value) =>
    `(title ~ "${value}" OR identifier_label ~ "${value}")`,
};

export default [rule];
