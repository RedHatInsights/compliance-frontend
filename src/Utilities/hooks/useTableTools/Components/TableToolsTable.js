import React from 'react';
import propTypes from 'prop-types';
import { Pagination, PaginationVariant } from '@patternfly/react-core';
import { Table, TableBody, TableHeader } from '@patternfly/react-table';
import PrimaryToolbar from '@redhat-cloud-services/frontend-components/PrimaryToolbar';
import TableToolbar from '@redhat-cloud-services/frontend-components/TableToolbar';
import useTableTools from '../useTableTools';

const TableToolsTable = ({
  items = [],
  columns = [],
  filters = [],
  options = {},
  toolbarProps: toolbarPropsProp,
  ...tablePropsRest
}) => {
  const { toolbarProps, tableProps, ColumnManager, TreeTableToggle } =
    useTableTools(items, columns, {
      filters,
      toolbarProps: toolbarPropsProp,
      tableProps: tablePropsRest,
      ...options,
    });

  return (
    <React.Fragment>
      <PrimaryToolbar {...toolbarProps}>
        {TreeTableToggle && <TreeTableToggle />}
      </PrimaryToolbar>

      <Table {...tableProps}>
        <TableHeader />
        <TableBody />
      </Table>

      {/* The -1 are to combat a bug currently in the TableToolbar component */}
      <TableToolbar isFooter results={-1} selected={-1}>
        {toolbarProps.pagination && (
          <Pagination
            variant={PaginationVariant.bottom}
            {...toolbarProps.pagination}
          />
        )}
      </TableToolbar>

      {ColumnManager && <ColumnManager />}
    </React.Fragment>
  );
};

TableToolsTable.propTypes = {
  items: propTypes.array.isRequired,
  columns: propTypes.arrayOf(
    propTypes.shape({
      title: propTypes.node,
      transforms: propTypes.array,
      sortByProperty: propTypes.string,
      sortByArray: propTypes.array,
      sortByFunction: propTypes.func,
      renderFunc: propTypes.func,
    })
  ).isRequired,
  filters: propTypes.object,
  options: propTypes.object,
  toolbarProps: propTypes.object,
};

export default TableToolsTable;
