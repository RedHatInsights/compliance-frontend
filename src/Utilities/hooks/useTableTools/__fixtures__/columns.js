import { severityLevels } from './items';

export default [
    {
        title: 'Name',
        sortByProperty: 'name',
        renderFunc: (_a, _b, item) => (
            item.name
        )
    }, {
        title: 'Description',
        sortByProperty: 'description'
    }, {
        title: 'Severity sorted by function',
        sortByFunction: (item) => (item.name),
        renderFunc: (_a, _b, item) => (
            item.severity
        )
    }, {
        title: 'Severity by Array',
        sortByProperty: 'severity',
        sortByArray: severityLevels,
        renderFunc: (_a, _b, item) => (
            item.severity
        )
    }
];
