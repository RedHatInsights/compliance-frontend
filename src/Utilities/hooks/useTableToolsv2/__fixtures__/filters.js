export const exampleFilters = [
  {
    type: 'text',
    label: 'Name',
  },
  {
    type: 'checkbox',
    label: 'Compliant',
    items: [
      { label: 'Compliant', value: 'true' },
      { label: 'Non-compliant', value: 'false' },
    ],
  },
  {
    type: 'checkbox',
    label: 'Systems meeting compliance',
    items: [
      { label: '90 - 100%', value: '90-100' },
      { label: '70 - 89%', value: '70-89' },
      { label: '50 - 69%', value: '50-69' },
      { label: 'Less than 50%', value: '0-49' },
    ],
  },
];

export default [
  {
    type: 'text',
    label: 'Name',
  },
  {
    type: 'hidden',
    label: 'Hidden filter',
  },
  {
    type: 'checkbox',
    label: 'Checkbox Filter',
    items: ['OPTION 1', 'OPTION 2', 'OPTION 3'].map((option) => ({
      label: option,
      value: `${option}-value`,
    })),
  },
  {
    type: 'radio',
    label: 'Radio Filter',
    items: ['OPTION 1', 'OPTION 2', 'OPTION 3'].map((option) => ({
      label: option,
      value: `${option}-value`,
    })),
  },
  {
    type: 'UNKNOWNTYPE',
    label: 'Invalid Filter',
    items: ['OPTION 1', 'OPTION 2', 'OPTION 3'].map((option) => ({
      label: option,
      value: `${option}-value`,
    })),
  },
  {
    type: 'group',
    label: 'Filter group',
    groupSelectable: true,
    items: [
      {
        label: 'Parent 1',
        value: 1,
        items: [
          { label: 'Child 1', value: 1 },
          { label: 'Child 2', value: 2 },
        ],
      },
      {
        label: 'Parent 2',
        value: 2,
        items: [
          { label: 'Parent 2 Child 1', value: 1 },
          { label: 'Parent 2 Child 2', value: 2 },
        ],
      },
    ],
  },
];
