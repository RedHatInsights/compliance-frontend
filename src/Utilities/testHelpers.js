import useFeature from 'Utilities/hooks/useFeature';
jest.doMock('Utilities/hooks/useFeature', () => jest.fn());

export const useWithFeature = (feature, tests) => {
  describe(`with ${feature} turned ON`, () => {
    beforeEach(() => {
      useFeature.mockImplementation((useFeatureFeature) => {
        console.log('YOU', useFeatureFeature);
        return useFeatureFeature === feature ? true : false;
      });
    });

    tests();
  });

  describe(`with ${feature} turned OFF`, () => {
    beforeEach(() => {
      useFeature.mockImplementation((useFeatureFeature) =>
        useFeatureFeature === feature ? false : false
      );
    });

    tests();
  });
};
