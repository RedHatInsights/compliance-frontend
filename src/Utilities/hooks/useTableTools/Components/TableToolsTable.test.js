import { render, screen, within } from '@testing-library/react';
import '@testing-library/jest-dom';

import items from '../__fixtures__/items';
import columns from '../__fixtures__/columns';
import filters from '../__fixtures__/filters';
import { filterHelpers } from '../testHelpers.js';
expect.extend(filterHelpers);

import TableToolsTable from './TableToolsTable';

describe('TableToolsTable', () => {
  const exampleItems = items(30).sort((item) => item.name);
  const defaultProps = {
    columns,
    'aria-label': 'Test Table',
    filters: { filterConfig: filters },
    items: exampleItems,
  };

  it('expect to have toolbar properly rendered', async () => {
    render(
      <TableToolsTable
        {...{
          ...defaultProps,
          options: {
            preselected: [exampleItems[1].id],
            selectedFilter: true,
            onSelect: () => {},
          },
        }}
      />
    );

    const toolbar = screen.getByLabelText('Table toolbar');
    await screen.findByLabelText('Table toolbar');

    expect(
      within(toolbar).getByLabelText('Conditional filter toggle')
    ).toBeInTheDocument();

    expect(
      within(toolbar).getByLabelText('kebab dropdown toggle')
    ).toBeInTheDocument();
    expect(within(toolbar).getByLabelText('Pagination')).toBeInTheDocument();
  });

  // TODO This can be properly implemented once on react 18 and newer RTL packages
  it.skip('expect to have filters properly rendered', () => {
    render(<TableToolsTable {...defaultProps} />);

    expect(screen.getByLabelText('Table toolbar')).toHaveFiltersFor(filters);
  });
});
