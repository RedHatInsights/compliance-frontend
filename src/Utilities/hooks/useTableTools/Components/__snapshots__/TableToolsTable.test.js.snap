// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolsTable expect to have filters properly rendered 1`] = `
<div
  class="pf-c-pagination__total-items"
>
  <b>
    1
     - 
    1
  </b>
   
  of
   
  <b>
    1
  </b>
   
  
</div>
`;

exports[`TableToolsTable expect to have filters properly rendered 2`] = `
<div
  class="pf-c-pagination__total-items"
>
  <b>
    1
     - 
    10
  </b>
   
  of
   
  <b>
    30
  </b>
   
  
</div>
`;

exports[`TableToolsTable expect to render without error 1`] = `
<Fragment>
  <PrimaryToolbar
    actionsConfig={
      Object {
        "actions": Array [
          undefined,
        ],
      }
    }
    activeFiltersConfig={
      Object {
        "filters": Array [],
        "onDelete": [Function],
      }
    }
    filterConfig={
      Object {
        "items": Array [
          Object {
            "filterValues": Object {
              "onChange": [Function],
              "value": undefined,
            },
            "id": "name",
            "label": "Name",
            "placeholder": "Filter by name",
            "type": "text",
          },
          Object {
            "filterValues": Object {
              "items": Array [
                Object {
                  "label": "OPTION 1",
                  "value": "OPTION 1",
                },
                Object {
                  "label": "OPTION 2",
                  "value": "OPTION 2",
                },
                Object {
                  "label": "OPTION 3",
                  "value": "OPTION 3",
                },
              ],
              "onChange": [Function],
              "value": undefined,
            },
            "id": "checkboxfilter",
            "label": "Checkbox Filter",
            "placeholder": "Filter by checkbox filter",
            "type": "checkbox",
          },
          Object {
            "filterValues": Object {
              "items": Array [
                Object {
                  "label": "OPTION 1",
                  "value": "OPTION 1",
                },
                Object {
                  "label": "OPTION 2",
                  "value": "OPTION 2",
                },
                Object {
                  "label": "OPTION 3",
                  "value": "OPTION 3",
                },
              ],
              "onChange": [Function],
              "value": undefined,
            },
            "id": "radiofilter",
            "label": "Radio Filter",
            "placeholder": "Filter by radio filter",
            "type": "radio",
          },
          Object {
            "className": undefined,
            "filterValues": Object {
              "groups": Array [
                Object {
                  "items": Array [
                    Object {
                      "label": "Child 1",
                      "type": "checkbox",
                      "value": 1,
                    },
                    Object {
                      "label": "Child 2",
                      "type": "checkbox",
                      "value": 2,
                    },
                  ],
                  "label": "Parent 1",
                  "type": "checkbox",
                  "value": 1,
                },
                Object {
                  "items": Array [
                    Object {
                      "label": "Parent 2 Child 1",
                      "type": "checkbox",
                      "value": 1,
                    },
                    Object {
                      "label": "Parent 2 Child 2",
                      "type": "checkbox",
                      "value": 2,
                    },
                  ],
                  "label": "Parent 2",
                  "type": "checkbox",
                  "value": 2,
                },
              ],
              "onChange": [Function],
              "selected": undefined,
            },
            "id": "filtergroup",
            "label": "Filter group",
            "type": "group",
          },
        ],
      }
    }
    pagination={
      Object {
        "itemCount": 30,
        "onPerPageSelect": [Function],
        "onSetPage": [Function],
        "page": 1,
        "perPage": 10,
      }
    }
  />
  <Table
    actionsMenuAppendTo="inline"
    aria-label="Test Table"
    borders={true}
    canCollapseAll={false}
    canSelectAll={true}
    canSortFavorites={true}
    cells={
      Array [
        Object {
          "exportKey": "id",
          "sortByProperty": "id",
          "title": "ID",
        },
        Object {
          "exportKey": "name",
          "renderFunc": [Function],
          "sortByProperty": "name",
          "title": "Name",
        },
        Object {
          "exportKey": "name",
          "renderExport": [Function],
          "sortByProperty": "name",
          "title": "Another Name column",
        },
        Object {
          "renderExport": [Function],
          "sortByProperty": "description",
          "title": "Description",
        },
        Object {
          "exportKey": "severity",
          "renderFunc": [Function],
          "sortByFunction": [Function],
          "title": "Severity sorted by function",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "exportKey": "severity",
          "renderFunc": [Function],
          "sortByArray": Array [
            "high",
            "medium",
            "low",
            "unknown",
          ],
          "sortByProperty": "severity",
          "title": "Severity by Array",
        },
      ]
    }
    className=""
    collapseAllAriaLabel=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    isHeaderSelectDisabled={false}
    isNested={false}
    isStickyHeader={false}
    isTreeTable={false}
    onCollapse={[Function]}
    onSort={[Function]}
    ouiaSafe={true}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Object {
          "cells": Array [
            Object {
              "title": 1,
            },
            Object {
              "title": "TEST ITEM #1",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "medium",
            },
            Object {
              "title": "medium",
            },
          ],
          "itemId": 1,
        },
        Object {
          "cells": Array [
            Object {
              "title": 2,
            },
            Object {
              "title": "TEST ITEM #2",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "low",
            },
            Object {
              "title": "low",
            },
          ],
          "itemId": 2,
        },
        Object {
          "cells": Array [
            Object {
              "title": 3,
            },
            Object {
              "title": "TEST ITEM #3",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "unknown",
            },
            Object {
              "title": "unknown",
            },
          ],
          "itemId": 3,
        },
        Object {
          "cells": Array [
            Object {
              "title": 4,
            },
            Object {
              "title": "TEST ITEM #4",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "high",
            },
            Object {
              "title": "high",
            },
          ],
          "itemId": 4,
        },
        Object {
          "cells": Array [
            Object {
              "title": 5,
            },
            Object {
              "title": "TEST ITEM #5",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "medium",
            },
            Object {
              "title": "medium",
            },
          ],
          "itemId": 5,
        },
        Object {
          "cells": Array [
            Object {
              "title": 6,
            },
            Object {
              "title": "TEST ITEM #6",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "low",
            },
            Object {
              "title": "low",
            },
          ],
          "itemId": 6,
        },
        Object {
          "cells": Array [
            Object {
              "title": 7,
            },
            Object {
              "title": "TEST ITEM #7",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "unknown",
            },
            Object {
              "title": "unknown",
            },
          ],
          "itemId": 7,
        },
        Object {
          "cells": Array [
            Object {
              "title": 8,
            },
            Object {
              "title": "TEST ITEM #8",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "high",
            },
            Object {
              "title": "high",
            },
          ],
          "itemId": 8,
        },
        Object {
          "cells": Array [
            Object {
              "title": 9,
            },
            Object {
              "title": "TEST ITEM #9",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "medium",
            },
            Object {
              "title": "medium",
            },
          ],
          "itemId": 9,
        },
        Object {
          "cells": Array [
            Object {
              "title": 10,
            },
            Object {
              "title": "TEST ITEM #10",
            },
            Object {
              "title": undefined,
            },
            Object {
              "title": "DESCRIPTION",
            },
            Object {
              "title": "low",
            },
            Object {
              "title": "low",
            },
          ],
          "itemId": 10,
        },
      ]
    }
    selectVariant="checkbox"
    sortBy={
      Object {
        "direction": "asc",
        "index": 1,
      }
    }
    variant={null}
  >
    <TableHeader />
    <TableBody />
  </Table>
  <TableToolbar
    isFooter={true}
    results={-1}
    selected={-1}
  >
    <Pagination
      className=""
      defaultToFullPage={false}
      firstPage={1}
      isCompact={false}
      isDisabled={false}
      isSticky={false}
      itemCount={30}
      itemsEnd={null}
      itemsStart={null}
      offset={0}
      onFirstClick={[Function]}
      onLastClick={[Function]}
      onNextClick={[Function]}
      onPageInput={[Function]}
      onPerPageSelect={[Function]}
      onPreviousClick={[Function]}
      onSetPage={[Function]}
      ouiaSafe={true}
      page={1}
      perPage={10}
      perPageComponent="div"
      perPageOptions={
        Array [
          Object {
            "title": "10",
            "value": 10,
          },
          Object {
            "title": "20",
            "value": 20,
          },
          Object {
            "title": "50",
            "value": 50,
          },
          Object {
            "title": "100",
            "value": 100,
          },
        ]
      }
      titles={
        Object {
          "currPage": "Current page",
          "items": "",
          "itemsPerPage": "Items per page",
          "ofWord": "of",
          "optionsToggle": "",
          "page": "",
          "pages": "",
          "paginationTitle": "Pagination",
          "perPageSuffix": "per page",
          "toFirstPage": "Go to first page",
          "toLastPage": "Go to last page",
          "toNextPage": "Go to next page",
          "toPreviousPage": "Go to previous page",
        }
      }
      variant="bottom"
      widgetId="options-menu"
    />
  </TableToolbar>
</Fragment>
`;
